src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -411,7 +450,11 @@ impl StaticHeader {
         // it must also have correct CRC, no weird stuff in fields,
         // etc!
         match StaticHeader::setup(f) {
-            Ok(Some(sh)) => Ok(Some((sh.pool_uuid, sh.dev_uuid))),
+            Ok(Some(SetupResult::Ok(sh))) => Ok(Some((sh.pool_uuid, sh.dev_uuid))),
+            Ok(Some(SetupResult::OkWithError(sh, err))) => {
+                warn!("StaticHeader::setup didn't completely succeed: {:?}", err);
```

Comment:
The I/O error that you picked up doesn't really explain what happened. So, a clearer explanation would be something like:

```warn!("Experienced an I/O error while attempting to repair an ill-formed, unreadable, or stale signature block: {:?}", err);```

This has the flaw that it does not identify which signature block, on which device, etc., so it is more alarming than helpful in that simple state. 

But you can improve it quite a bit, by including the good StaticHeader, the one that was succesfully read, in the error message (or possibly just selected identifying information from that StaticHeader). This would probably be worth a separate function, since you do it in two places.

Reply:
I extended your text a bit. Tell me if it's what you had in mind otherwise I'll just update. :)